src/Microsoft.AspNetCore.Mvc.Core/Formatters/TextInputFormatter.cs

```diff
@@ -82,23 +129,30 @@ protected Encoding SelectCharacterEncoding(InputFormatterContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
-            if (SupportedEncodings?.Count == 0)
+            if (SupportedEncodings.Count == 0)
             {
-                var message = Resources.FormatTextInputFormatter_SupportedEncodingsMustNotBeEmpty(nameof(SupportedEncodings));
+                var message = Resources.FormatTextInputFormatter_SupportedEncodingsMustNotBeEmpty(
+                    nameof(SupportedEncodings));
                 throw new InvalidOperationException(message);
             }
 
-            var request = context.HttpContext.Request;
-
-            var requestEncoding = request.ContentType == null ? null : MediaType.GetEncoding(request.ContentType);
```

Comment:
Shouldn't we fix MediaType.GetEncoding?

I would imagine we just put the hardcoded list of aliases in there and we check it before calling Encoding.GetEncoding (returning null otherwise).

https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/Formatters/MediaType.cs#L329